test/visual-diff/visual-tests

```diff
@@ -567,6 +567,26 @@
         ".i-amphtml-story-no-rotation-overlay"
       ]
     },
+    {
+      "url": "examples/visual-tests/amp-inabox/amp-ads.html",
+      "name": "AMP Inabox Ad",
+      "loading_complete_css": [
+        ".slot-render-ended",
+        ".impression-viewable",
+        ".slot-onload",
+        ".slot-visibility-changed",
+      ],
+      // we want to wait for the active view event to be fired
+      "loading_complete_delay_ms": 3000,
+    },
+    {
+      "url": "examples/visual-tests/amp-inabox/amp-ads-second.html",
+      "name": "AMP Inabox Two Ads",
+      "forbidden_css": [".second-impression-viewable"],
```

Comment:
"forbidden_css" means makes the test fail if the class is ever found on the page. Depending on what you're trying to achieve, you might want to do "loading_incomplete_css" - the latter allows a class to be on the page at first but will reject the test if the class stays visible for long (you might have actually meant to use "forbidden_css" and I just don't understand the flow of the test)

Reply:
So within the test's script ".second-impression-viewable" will be set if the second ad is determined to be viewed (shows up on the viewport for more than 1 second); it does not exist before then. The test page is written so that the second ad is well out of the viewport (which considering your other comments I should set an explicit size for), so without the custom test to scroll down it would never be set.

I think for this case forbidden_css is the right one to use; do let me know your opinion on the matter.

Reply:
Though now that you mentioned it having flags that will be _unset_ when the ad is viewed alongside the flags that will be set is the better way to do this. I will change the code and you can get back to it.

Reply:
In that case, forbidden_css is fine!